include/aws/common/task_scheduler.h

```diff
@@ -33,7 +33,12 @@ struct aws_task {
     struct aws_linked_list_node node;
     struct aws_priority_queue_node priority_queue_node;
     const char *type_tag;
-    size_t reserved;
+
+    /* honor the ABI compat */
+    union {
+        bool scheduled;
```

Comment:
Do you want to bitfield this?

Reply:
We can, but we don’t have to til we start using the other 3 bytes.